types/yup/index.d.ts

```diff
@@ -431,15 +431,21 @@ export interface FormatErrorParams {
 export type LocaleValue = string | ((params: FormatErrorParams) => string);
 
 type MessageFromParameters<P extends unknown[]> = {
-    [K in keyof P]: P[K] extends TestOptionsMessage<any> ? P[K] : never;
+    [K in keyof P]: Extract<P[K], TestOptionsMessage<any>> extends TestOptionsMessage<any>
+        ? Extract<P[K], TestOptionsMessage<any>>
+        : never;
```

Comment:
Any way you can demonstrate this in a test? The difference is not clear to me

Reply:
The only reason I needed this was because of the union in `string.matches`. Without the `Extract`, it would fail because that argument doesn't match only `TestOptionsMessage`. I'm not sure how to show that in a test besides somehow negating the change. This is part of why I'm considering inverting the logic and making the message argument in methods pull from `LocaleObject` instead of building `LocaleObject` from the schemas.

Reply:
ok